### PR TITLE
Fixes #11936 - Allow authenticating with kerberos

### PR DESCRIPTION
**Pull request by adamruzicka**
For now it requires `net-ssh-krb` to be required into the `smart-proxy` or `smart_proxy_dynflow_core`

Packaging PRs:
- [DEB](https://github.com/theforeman/foreman-packaging/pull/1664)
- [RPM](https://github.com/theforeman/foreman-packaging/pull/1665)

### CODE REVIEW

**Comment by iNecas** [comment]
I guess there should be also a PR to https://github.com/theforeman/smart_proxy_remote_execution_ssh/blob/master/settings.d/remote_execution_ssh.yml.example, to mention the config option

File: lib/foreman_remote_execution_core/script_runner.rb [hunk 1]
```diff
@@ -279,5 +282,17 @@ def check_expecting_disconnect
         @expecting_disconnect = true
       end
     end
+
+    def available_authentication_methods
+      methods = %w(pubkey) # Always use pubkey auth as fallback
+      if settings.kerberos_auth
```
Comment (iNecas) [inline lib/foreman_remote_execution_core/script_runner.rb]: I'm getting `NoMethodError - undefined method `kerberos_auth' for #<Hash:0x005609203c17f0>`, tried with smart proxy 1.15 as well as develop, with latest versions of plugins. Am I missing something?

**Comment by adamruzicka** [comment]
Had it prepared, forgot to open the [PR](https://github.com/theforeman/smart_proxy_remote_execution_ssh/pull/33).

**Comment by iNecas** [comment]
After workarounding the issue with hash[], it was working nicely. Ready to merge once this issue is clarified

File: lib/foreman_remote_execution_core/script_runner.rb (hunk 1 above)
Comment (adamruzicka) [inline lib/foreman_remote_execution_core/script_runner.rb]: Do you run smart_proxy_dynflow_core as a separate process or as a part of smart proxy? From what I remember I expected `settings` to be an `OpenStruct` whereas in your case it is a `Hash`. Will investigate tomorrow

**Comment by adamruzicka** [comment]
Comments addressed

**Comment by iNecas** [comment]
Seems like Jenkis is having some issues, let's wait till it's operating again

**Comment by adamruzicka** [comment]
[test]

File: lib/foreman_remote_execution_core/script_runner.rb [hunk 2]
```diff
@@ -1,4 +1,9 @@
 require 'net/ssh'
+begin
+  require 'net/ssh/krb'
+rescue LoadError
+  $stderr.puts 'Failed to load net/ssh/krb, kerberos authentication won\'t be available'
```
Comment (iNecas) [inline lib/foreman_remote_execution_core/script_runner.rb]: One last thing: for the users that don't plan to use keberos authentication, this produces an error message every time they start the service. I guess we could just be silent here and include the cause of the error ('net/ssh/krb' missing, in the log message when `Net::SSH::Kerberos` is not authenticated.

**Comment by iNecas** [comment]
Thanks @adamruzicka

**Comment by iNecas** [comment]
I guess we also need an installer update

**Comment by adamruzicka** [comment]
It's in the works https://github.com/theforeman/puppet-foreman_proxy/pull/351

**Pull request merged**